File: Src/Services/Identity/UserValidator.cs
using Microsoft.AspNetCore.Identity;
using MusicPlayerBackend.Data.Entities;

namespace MusicPlayerBackend.Services.Identity;

public sealed class UserValidator : IUserValidator<User>
{
    public Task<IdentityResult> ValidateAsync(UserManager<User> manager, User user)
    {
        return Task.FromResult(IdentityResult.Success);
    }
}

File: Src/MusicPlayerBackend.Identity/UserValidator.fs
namespace MusicPlayerBackend.Identity

open System.Threading.Tasks
open Microsoft.AspNetCore.Identity
open MusicPlayerBackend.Data.Entities

type UserValidator() =
    interface IUserValidator<User> with
        member this.ValidateAsync(_, _) =
            Task.FromResult(IdentityResult.Success)
